lib/fakeBlockChain.js

```diff
@@ -2,9 +2,16 @@ const Buffer = require('safe-buffer').Buffer
 const utils = require('ethereumjs-util')
 
 module.exports = {
-  getBlock: function (n, cb) {
-    // FIXME: this will fail on block numbers >53 bits
-    var hash = utils.sha3(Buffer.from(utils.bufferToInt(n).toString(), 'utf8'))
+  getBlock: function (blockTag, cb) {
+    var hash
+
+    if (Buffer.isBuffer(blockTag)) {
+      hash = utils.sha3(blockTag)
+    } else if (Number.isInteger(blockTag)) {
+      hash = utils.sha3(utils.toBuffer(hash).toStirng('hex'))
```

Comment:
Seemingly this code path is never triggered given it succeeded with `toStirng`.

Reply:
I'm not really getting this comment, and now you want to leave this as ``toStirng`` ?? :-)

Reply:
``utils.toBuffer(hash)`` is obviously wrong.

Reply:
And I think the whole string from the corrected ``utils.toBuffer(hash).toStirng('hex')`` has to be prefixed with ``0x``, otherwise this will deliver a false result (``utils.sha3`` assuming to be provided with a base-10 integer string).

Reply:
Although this might be never triggered within library context, it probably makes sense to leave the different paths since it resembles now the API/usage of the corresponding function on ``ethereumjs-blockchain``.

Reply:
The comment meant that codepath is never triggered by tests and as such missed the broken code `toStirng`, which should be `toString`.

Reply:
fakeBlockchain is only used by state tests and state tests aren't using any of the opcodes involving block information.  Hence 'getBlock' is never called.

Reply:
@axic Will you repair this now or are we leaving it "as is"? :-)

Reply:
It should repaired :)

Reply:
@holgerd77 @jwasinger updated both typos